### ISSUE TITLE
jira/notify fails on macOS executor

### ISSUE THREAD

**Issue by kevin--** [CircleCI-Public/jira-orb#18]
Using JIRA orb 2.0.1. on a macOS executor `macos.x86.medium.gen2`

Consistently errors out. Works fine on our windows executors

The circleci_jira.log is empty

## Error message
```
/bin/bash: -c: line 74: conditional binary operator expected

Exited with code exit status 2
```

## Root Cuase
This is because mac is using ZSH which does not support `-v`. We could use `-z` for this perhaps as it's meaning is similar and exists on both interpreters

## Potential Fix
BASH:
```
-z string
True if the length of string is zero.
```
ZSH:
```
-z string
Check if a string is empty or not defined
```

**Comment by KyleTryon** [comment]
This mainly stems from the use of associative arrays `declare -A seen` which were added in [bash version 4](https://tldp.org/LDP/abs/html/bashver4.html), while it seems even the latest available versions of MacOS are shipping with `3.2`

```
static:~ distiller$ sw_vers
  ProductName:            macOS
  ProductVersion:         13.5.1
  BuildVersion:           22G90

static:~ distiller$ bash --version
  GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin22)
  Copyright (C) 2007 Free Software Foundation, Inc.
static:~ distiller$ zsh --version
  zsh 5.9 (x86_64-apple-darwin22.0)
```

The issue is therefore unfortunately not solved by #15 alone. 

I did some checking and reached out to the MacOS team to confirm, and we _are_ using bash by default. So, we should be able to solve this by switching to ZSH at the appropriate time.

I am attempting this fix now but currently running into some issues, I will sort this out though.

**Comment by kevin--** [comment]
Thanks for the update @KyleTryon 
Are you suggesting that if we run the job with `shell: zsh` it may succeed?

**Comment by KyleTryon** [comment]
@kevin-- if you were to set the job shell to ZSH, this may work but could adversely effect other scripts you may run as a part of the job.

I am looking to see if we could dynamically switch to ZSH, given the limitations we have with not actually shipping code/scripts.

**Comment by KyleTryon** [comment]
Thank you all for the information! This ended up being a first complex issue that was solved with a fairly simple change. See #26 for the how-it-was-done

For those just looking for this fix, you can now upgrade to `2.0.2`

- https://circleci.com/developer/orbs/orb/circleci/jira?version=2.0.2
- https://github.com/CircleCI-Public/jira-orb/releases/tag/v2.0.2
